lib/Commands/InterruptSignalAwaiter.cpp

```diff
@@ -0,0 +1,88 @@
+//===-- InterruptSignalAwaiter.cpp ----------------------------------------===//
```

Comment:
I would put this in `Basic` since it has no higher dependencies, and is general purpose infrastructure.